onnxscript/rewriter/generic_pattern.py

```diff
@@ -294,6 +294,78 @@ def make_node(
         return proto
 
 
+class PatternMatchResult:
+    """Stores information about a match if a match was successful."""
+
+    def __init__(
+        self,
+        pattern: "GenericPattern",
+        model_nodes: typing.Sequence[oir.Node],
+        pattern_nodes: typing.Sequence[oir.Node],
+        pattern_input_names: typing.Sequence[str],
+        pattern_output_names: typing.Sequence[str],
+    ):
+        assert len(model_nodes) == len(pattern_nodes), (
```

Comment:
Is this an error that can be caused by user code? If so I would raise a ValueError instead of asserting.

Reply:
No it cannot be created by the user.

Reply:
Or maybe call out in the assert string that this would be a bug?

Reply:
I'll remove it. I prefer long error messages but that's your call.